theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
check the file cuda/opt.py We reuse the same op on the GPU.

On Tue, Mar 8, 2016 at 2:44 PM, Xavier Bouthillier <notifications@github.com
> wrote:

> In theano/tensor/nlinalg.py
> <https://github.com/Theano/Theano/pull/4169#discussion_r55415759>:
>
> > @@ -179,6 +185,10 @@ class ExtractDiag(Op):
> >      __props__ = ("view",)
> >
> >      def __init__(self, view=False):
> > +        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
>
> @nouiz <https://github.com/nouiz> It says ExtractDiag works on the GPU
> but I can't find its GPU implementation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/4169/files#r55415759>.
>